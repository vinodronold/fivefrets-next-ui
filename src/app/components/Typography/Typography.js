import React, { PureComponent } from 'react'

export const GetLetterSpacing = ls => (typeof ls === 'number' ? `${ls}rem` : ls)

export default class Typo extends PureComponent {
  render() {
    const { fs, fw, ls, lh } = this.props
    return (
      <div>
        {this.props.children}
        <style jsx>{`
          div {
            text-decoration: inherit;
            text-transform: inherit;
          }
        `}</style>
        <style jsx>{`
          div {
            font-size: ${fs}rem;
            font-weight: ${fw};
            letter-spacing: ${GetLetterSpacing(ls)};
            line-height: ${lh}rem;
          }
        `}</style>
      </div>
    )
  }
}

export const GenerateTypo = (fs, fw, ls, lh) => ({ children }) => (
  <Typo fs={fs} fw={fw} ls={ls} lh={lh}>
    {children}
  </Typo>
)

export const Display1 = GenerateTypo(2.125, 400, 'normal', 2.5)
export const Caption = GenerateTypo(0.75, 400, 0.08, 1.25)
