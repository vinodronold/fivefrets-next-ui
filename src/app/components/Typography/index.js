import { Caption, Display1 } from './Typography'

export { Caption, Display1 }
