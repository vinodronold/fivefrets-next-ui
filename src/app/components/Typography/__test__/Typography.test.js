import React from 'react'
import renderer from 'react-test-renderer'
import { GetLetterSpacing, Display1, Caption } from '../Typography'

describe('component/Typography', () => {
  it('GetLetterSpacing should return rem when letter-spacing is number', () => {
    expect(GetLetterSpacing(1)).toBe('1rem')
  })

  it('GetLetterSpacing should return same string when letter-spacing is string', () => {
    let _ls = 'normal'
    expect(GetLetterSpacing(_ls)).toBe(_ls)
  })

  it('Display1 should match Snapshot', () => {
    const _display_1 = renderer.create(<Display1>Display1</Display1>)
    expect(_display_1).toMatchSnapshot()
  })

  it('Caption should match Snapshot', () => {
    const _caption = renderer.create(<Caption>Caption</Caption>)
    expect(_caption).toMatchSnapshot()
  })
})
