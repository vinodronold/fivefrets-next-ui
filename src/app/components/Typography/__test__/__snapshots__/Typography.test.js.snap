// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/Typography Caption should match Snapshot 1`] = `
<div
  className="jsx-4247411757 jsx-3754337244"
>
  Caption
</div>
`;

exports[`component/Typography Display1 should match Snapshot 1`] = `
<div
  className="jsx-4247411757 jsx-2107752004"
>
  Display1
</div>
`;
